src/ansible_navigator/runner/base.py

```diff
@@ -0,0 +1,218 @@
+        if (
+            self._rotate_artifacts is not None
+            and self._runner_artifact_dir
+            and os.path.exists(self._runner_artifact_dir)
+        ):
+
+            self._logger.debug(
+                "delete ansible-runner artifact directory at path %s", self._runner_artifact_dir
+            )
+            shutil.rmtree(self._runner_artifact_dir, ignore_errors=True)
+
+    @staticmethod
+    def _generate_tmp_directory():
+        """generate a tmp directory for artifacts"""
+        return tempfile.mkdtemp(prefix="ansible-navigator_")
+
+    def _set_private_data_directory(self, provided: Union[str, None]) -> None:
+        """Set the private data directory to the provided, the username, or a uuid"""
+
+        if isinstance(provided, str):
+            if os.access(provided, os.W_OK | os.R_OK | os.X_OK):
+                private_data_directory = provided
+                source = "user provided"
+            else:
+                self._logger.debug("Provided private data dir `%s` was not user writable")
+                private_data_directory = self._generate_tmp_directory()
+                source = "user provided, but changed to tmp location due to permissions"
+        else:
+            private_data_directory = self._generate_tmp_directory()
+            source = "not user provided, used tmp location"
+
+        self._private_data_dir = private_data_directory
+        self._logger.debug("private data dir %s: %s", source, self._private_data_dir)
+
+    def runner_artifacts_handler(self, artifact_dir):
+        """
+        ansible-runner callback to handle artifacts after each runner innvocation
+
+        Args:
+            artifact_dir ([str]): The directory path of artifact directory for current \
+                                  runner invocation.
+        """
+        self._logger.debug("ansible-runner artifact_dir set to: '%s'", artifact_dir)
+        self._runner_artifact_dir = artifact_dir
+
+    def runner_cancelled_callback(self):
+        """check by runner to see if it should cancel"""
+        return self.cancelled
+
+    def runner_finished_callback(self, runner: Runner):
+        """called when runner finishes
+
+        Args:
+            runner (Runner): a runner instance
+        """
+        self.status = runner.status
+        self.finished = True
+
+    def _add_env_vars_to_args(self):
+        self._runner_args["envvars"] = {
+            k: v for k, v in os.environ.items() if k.startswith("ANSIBLE_")
+        }
+        self._runner_args["envvars"].update(self._set_environment_variable)
+        for env_var in self._pass_environment_variable:
+            value = os.environ.get(env_var)
+            if value is None:
+                self._logger.warning(
+                    "Pass through environment variable `%s`" " not currently set, discarded",
+                    env_var,
+                )
+            else:
+                self._runner_args["envvars"][env_var] = value
```

Comment:
A more semantically correct version of this would be
```suggestion
            try:
                env_var_value = os.environ[env_var]
            except KeyError:
                self._logger.warning(
                    "Pass through environment variable `%s`" " not currently set, discarded",
                    env_var,
                )
            else:
                self._runner_args["envvars"][env_var] = env_var_value
```

Reply:
I agree this a viable alternative, although the current implementation is fine and this falls outside the scope of the PR.

LMK if you want to PR if it is something you would like to change,